airflow/providers/microsoft/azure/hooks/azure_functions.py

```diff
@@ -0,0 +1,113 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from typing import Any
+
+import requests
+from azure.identity import ClientSecretCredential
+
+from airflow.providers.http.hooks.http import HttpHook
+from airflow.providers.microsoft.azure.utils import get_field
+
+
+class AzureFunctionsHook(HttpHook):
```

Comment:
Should we subclass AzureBaseHook? See https://learn.microsoft.com/en-us/python/api/azure-functions/azure.functions?view=azure-python

Reply:
we discussed this earlier with @josh-fell and concluded that we'll use HttpHook as we are doing a http based call in order to invoke Azure Functions App

Reply:
This would require users to set a different connection instead of using the already set connection for Base Azure hook. 
For me, unless we cannot invoke the azure functions using the base hook, then we can use HTTP hook but I doubt if we cannot.

Reply:
Can you please refer to this PR comment from Josh [here](https://github.com/apache/airflow/pull/28485/files#r1069615358)  and let me know your opinion